### PR TITLE
QOL, Fixes and new Shortcuts

### PR DESCRIPTION
**Pull request by RapidThrower264**
- Adds new parameter (handle line breaks) which can be set to disable the automatic line break before the item's type
  - Ideally prevents the need to do "custom item type" handling when trying to put `*Soulbound*` or others at the end
- Moves Stat parsing away from `StringColorParser`
- Improved color code readability when there is an invalid character code
- Fixes special flags for the `REQUIRE` stat to being placed after the text
- Implements secondary color (could be implemented wrong)
- Added new Stats (shortcuts)
  - `REFORGEABLE` - `%%DARK_GRAY%%This item can be reforged!`
  - `ITEM_STAT_RED`, `ITEM_STAT_GREEN`, `ITEM_STAT_PURPLE`
     - Creates an item stat with the stat, color and amount. Requires 2 more parameters.
     - Usage: `%%ITEM_STAT_RED:<Stat>:<Amount>%%` turns into `%%GRAY%%<Stat>: %%RED%%<Amount>%%GRAY%%`

### CODE REVIEW

File: src/main/java/net/hypixel/nerdbot/command/ItemGenCommand.java [hunk 1]
```diff
@@ -36,7 +36,8 @@ public void askForInfo(
             @AppOption(description = "The name of the item") String name,
             @AppOption(description = "The rarity of the item", autocomplete = "rarities") String rarity,
             @AppOption(description = "The description of the item") String description,
-            @Optional @AppOption(description = "The type of the item") String type
+            @Optional @AppOption(description = "The type of the item") String type,
+            @Optional @AppOption(description = "If you will handle line breaks at the end of the item's description") String handleLineBreaks
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/command/ItemGenCommand.java:40]: Change this to a boolean

File: src/main/java/net/hypixel/nerdbot/command/ItemGenCommand.java [hunk 2]
```diff
@@ -79,11 +80,17 @@ public void askForInfo(
 
         // writing the rarity if the rarity is not none
         if (itemRarity != Rarity.NONE) {
+            // checks if there is a type for the item
             if (type == null || type.equalsIgnoreCase("none")) {
                 type = "";
             }
+            // checking if there is custom line break happening
+            if (handleLineBreaks == null || !handleLineBreaks.equalsIgnoreCase("true")) {
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/command/ItemGenCommand.java:88]: With the above change
```suggestion
            if (!handleLineBreaks) {
```

**Comment by Aerhhh** [comment]
@RapidThrower24 it seems that stats don't have the colour applied to them, is it meant to be like this?
image

**Review by Aerhhh — CHANGES_REQUESTED** [review]

**Comment by RapidThrower264** [comment]
> @RapidThrower264 it seems that stats don't have the colour applied to them, is it meant to be like this?
> 
> The description is `%%PET_LUCK:1030%%` in the example

Some stats had secondary colors (`PET_LUCK`, `ATTACK_SPEED`, `HEALTH` and `SPEED`) included with then, but there was no implementation which used them.

File: src/main/java/net/hypixel/nerdbot/generator/StatColorParser.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
+package net.hypixel.nerdbot.generator;
+
+import net.hypixel.nerdbot.util.skyblock.Stat;
+
+public class StatColorParser {
+    /**
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/generator/StatColorParser.java:6]: Being nitpicky but put a blank line here

**Review by Aerhhh — CHANGES_REQUESTED** [review]

**Review by Aerhhh — APPROVED** [review]

**Pull request merged**